### PR TITLE
feat: refactoring and tests

### PR DESCRIPTION
**Pull request by themr0c**
TLDR: Fewer rules, all tested. Reference guide less overwhelming.

feat: consolidate the rules in fewer files (25 rules rather than 295)
feat: remove rules that are not defined in supplementary style guide or underlying style guides
feat: order fields consistently in rules
feat: point `link:` in rules to vale-at-red-hat docs
feat: point `source:` in rules to supplementary style guide
feat: create `tools/generate_vale_rule_tests.sh`
feat: create `tools/test_vale_rules.sh`
feat: generate vale rule tests
feat: generate reference guide and vale rule tests during preview
feat: refactor build script
feat: author extensive tests for all rules
feat: all rules pass the tests

### CODE REVIEW

**Comment by themr0c** [comment]
Status: The alerts are not the same (testing che-docs master branch). Need more in depth tests.

main:

✖ 128 errors, 502 warnings and 3200 suggestions in 764 files.

this branch:

✖ 217 errors, 40 warnings and 9769 suggestions in 764 files.

**Comment by themr0c** [comment]
After updates: testing che-docs 

* main: ✖ 3 errors, 502 warnings and 3230 suggestions in 764 files.
* this branch: ✖ 2 errors, 33 warnings and 3625 suggestions in 764 files.

**Comment by themr0c** [comment]
Now we know where we stand:

```
./tools/run_test_suite.sh 
ERROR: 1 in .vale/fixtures/RedHat/ExpressionsErrors/testinvalid.adoc / .vale/styles/RedHat/ExpressionsErrors.yml
ERROR: 1 in .vale/fixtures/RedHat/Usage/testinvalid.adoc / .vale/styles/RedHat/Usage.yml
ERROR: 1 in .vale/fixtures/RedHat/Using/testinvalid.adoc / .vale/styles/RedHat/Using.yml
ERROR: 1 in .vale/fixtures/RedHat/Which/testinvalid.adoc / .vale/styles/RedHat/Which.yml
4 tests to fix
```

**Comment by themr0c** [comment]
I have some doubt about raising an error for any occurrence of `which` not preceded by a comma.

**Review by aireilly — COMMENTED** [review]
LGTM
